src/Model/Order/Product/AssocVxc.php

```diff
@@ -38,7 +38,7 @@ class AssocVxc
      * @param \Megaport\Model\Order\Product\ConnectableProduct $bEnd
      * @param int $rateLimit
      */
-    public function __construct(string $productName, ConnectableProduct $aEnd, ConnectableProduct $bEnd, $rateLimit = 550)
+    public function __construct(string $productName, PortVlan $aEnd, CloudVlan $bEnd, $rateLimit = 550)
```

Comment:
Why make this so specific. The AssocVxc was created so that it can be used in Port to Port, Port to Cloud, Cloud to Cloud or anything else.

Reply:
Hm right now the constructor calls ```$aEnd->getVlan()``` which is only implemented in the PortVlan class and not in the ConnectableProduct interface, so my IDE was already complaining about it so I thought why not make it clear for the consumer of the class for now. But I agree it would be better to make it more generic in the end

Reply:
I changed the second argument back to ConnectableProduct, kept the first one PortVlan because of the implementation. Thanks

Reply:
You are right about the cloud to cloud vlans. I do not believe that product exists. So given the fact that you always start the vxc from a Port this looks good.